CIP-0030/README.md

```diff
@@ -200,13 +200,16 @@ Upon start, dApp can explicitly request a list of additional functionalities the
 
 DApps are expected to use this endpoint to perform an initial handshake and ensure that the wallet supports all their required functionalities. Note that it's possible for two extensions to be mutually incompatible (because they provide two conflicting features). While we may try to avoid this as much as possible while designing CIPs, it is also the responsability of wallet providers to assess whether they can support a given combination of extensions, or not. Hence wallets aren't expected to fail should they not recognize or not support a particular combination of extensions. Instead, they should decide what they enable and reflect their choice in the `cardano.{walletName}.extensions` field of the Full API. As a result, dApps may fail and inform their users or may use a different, less-efficient, strategy to cope with a lack of functionality.
 
-Extension's endpoints should be namespaced by `.cipXXXX.` with the returned `API` object, without any leading zeros. For example; CIP-0123's endpoints should be accessed by:
+It is at the extension author's discretion if they wish to separate their endpoints from the base API via namespacing. Although, it is highly recommend that authors do namespace all of their extensions. If namespaced, endpoints must be preceded by `.cipXXXX.` from the `API` object, without any leading zeros.
 
+For example; CIP-0123's endpoints should be accessed by:
 ```ts
 api.cip123.endpoint1()
 api.cip123.endpoint2()
 ```
 
+Authors should be careful when omitting namespacing. Omission should only be considered when creating endpoints to override those defined in this specification or other extensions. Even so when overriding; the new functionality should not prevent dApps from accessing past functionality thus overriding must ensure backwards compatibility.
+
```

Comment:
I would say that the extension should in this case contain a rationale and justification why its preferable to over-ride the pre-existing endpoint, and a clear note about backwards compatibility.  

Just to ensure the author has actually considered it.  It shouldn't be implied if I over-ride a pre-existing endpoint that I did this reasoning.  Then during review, if there is no reasoning its easy just to ask for it.  It wouldn't need to be exhaustive, for example with my earlier `signTx` example one could simply say:

```
This extension simply adds new ledger transactions to the set of valid transactions accepted by `signTx`.  It's backward compatible with the existing behavior.  
Wallets do not need to switch the set of valid transaction types dynamically when enabling this CIP, they can always accept the new transaction types with `signTx` whether this CIP is enabled explicitly or not. 
dApps can determine if this set of transactions are valid for `signTx` by enabling this extension.
```

Now it's clear the Author considered the issue, and the wallets and dapps have a clear direction for how they can implement it.

Reply:
Good call, added: [ba02ab5](https://github.com/cardano-foundation/CIPs/pull/577/commits/ba02ab514ff5707521d6c0417691e7b295aa9b8d).